benchmarks/benchmarks/pybench/persistence/database.py

```diff
@@ -0,0 +1,116 @@
+from typing import Sequence
+
+from pysqlite3 import Connection, connect
+
+from benchmarks.pybench.persistence.common import BenchAccessProto, BenchOperator, BenchRecord
+
+DATABASE_FILE_NAME = "adaptix_bench.db"
+
+
+class RecordNotFound(Exception):
+    def __init__(self, gid: str, name: str, sub_name: str):
+        self.gid = gid
+        self.name = name
+        self.sub_name = sub_name
+
+    def __str__(self):
+        return f"""Record not found for {self.name}/{self.sub_name}({self.gid})."""
+
+
+class IndexNotFound(Exception):
+    def __init__(self, hub_key: str):
+        self.hub_key = hub_key
+
+    def __str__(self):
+        return f"""Index not found - {self.hub_key}"""
+
+
+def migrate_sqlite(database_name: str):
+    create_table_ddl_q = """CREATE TABLE IF NOT EXISTS bench (
+    is_actual BOOLEAN,
+    benchmark_name TEXT,
+    benchmark_subname TEXT,
+    base TEXT,
+    local_id TEXT,
+    global_id TEXT,
+    tags TEXT,
+    kwargs TEXT,
+    distributions TEXT,
+    data TEXT,
+    created_at DATETIME
+    );"""
+    unique_index_ddl_q = """CREATE UNIQUE INDEX IF NOT EXISTS bench_unique ON
+     bench(benchmark_name, benchmark_subname, global_id) WHERE is_actual=TRUE;"""
+    with connect(database_name) as con:
+        cursor = con.cursor()
+        cursor.execute(create_table_ddl_q)
+        cursor.execute(unique_index_ddl_q)
+        con.commit()
+        cursor.close()
+
+
+class SQLite3BenchOperator(BenchOperator):
+    GET_BENCH_DATA_Q = """SELECT data, max(created_at) FROM bench WHERE benchmark_name = ?
+            AND benchmark_subname = ? and global_id = ?;"""
+    GET_INDEX_Q = """SELECT data, max(created_at) FROM bench_index WHERE hub_key = ?;"""
+
+    def __init__(self, accessor: BenchAccessProto | None, db_name: str):
+        self.accessor = accessor
+        self.db_name = db_name
+
+    def read_schemas_content(self) -> Sequence[str]:
```

Comment:
Schema describes benchmark case. There is no content of schema. This method should be called like `read_benchmark_results`